### PR TITLE
DOC: Added sentence to show that unmatched values return -1.

### LINKED ISSUE #38482: DOC: get_indexer returns non-matching with -1 positional
[An answer on my recent SO post](https://stackoverflow.com/a/65085536/4107349) used `get_indexer`, which worked great, but I soon learned if the indexer doesn't find a match it seems to use the last entry without raising an error:

```
df = pd.DataFrame({"ID": [6, 2, 4],
                   "to ignore": ["foo", "whatever", "idk"],
                   "value": ["A", "B", "asdf"],
                   })

df2 = pd.DataFrame({"ID_number": [1, 2, 3, 4, 5, 6],
                    "A": [0.91, 0.42, 0.85, 0.84, 0.81, 0.88],
                    "B": [0.11, 0.22, 0.45, 0.38, 0.01, 0.18],
                    })

df2 = df2.set_index('ID_number')
df['new_col'] = df2.values[df2.index.get_indexer(df['ID']), df2.columns.get_indexer(df['value'])]
```

I presumed the row with "asdf" would have raised an error, not returned the value for "B". This was problematic because I unwittingly processed data incorrectly and I enjoy being employed.

My interpretation of the [documentation](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Index.get_indexer.html) was that if `method` was not supplied it would be "default: exact matches only". Supplying `method = 'default'` and `tolerance = 0` was also not accepted when not using another method.

Also, I'm no serious programmer so I have likely misunderstood something and am only trying to help. Please feel free to correct / tell me to go away

### PR DESCRIPTION
**Pull request by DhruvBShetty**
- [x] closes #38482
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [x] whatsnew: Added sentence to show that unmatched values return -1 in get_indexer method

In [1]: python make.py --single pandas.Index.get_indexer
Out [1]:image

In [2]: pre-commit run --files pandas/core/indexes/base.py
Out [2] : image

### CODE REVIEW

**Review by MarcoGorelli — APPROVED** [review]
Thanks @DhruvBShetty

File: pandas/core/indexes/base.py [hunk 1]
```diff
@@ -3484,6 +3484,9 @@ def get_loc(self, key, method=None, tolerance=None):
         indexer should be then used as an input to ndarray.take to align the
         current data to the new index.
 
+        Returns -1 for unmatched values, for further explanation see the example
```
Comment (jreback) [inline pandas/core/indexes/base.py:3487]: move to the Notes section

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by MarcoGorelli** [comment]
Hi @DhruvBShetty can you address the comment please

**Comment by DhruvBShetty** [comment]
> This pull request is stale because it has been open for thirty days with no activity. Please update or respond to this comment if you're still interested in working on this.
Interested

**Comment by DhruvBShetty** [comment]
> This pull request is stale because it has been open for thirty days with no activity. Please update or respond to this comment if you're still interested in working on this.

Interested in finishing this.

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
please check the [pandas docstring guide](https://pandas.pydata.org/pandas-docs/stable/development/contributing_docstring.html) for how to write the "notes" section

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
Can you run the [validating docstrings](https://pandas.pydata.org/docs/development/contributing_documentation.html#updating-a-pandas-docstring) script for this function?

**Comment by DhruvBShetty** [comment]
> Can you run the [validating docstrings](https://pandas.pydata.org/docs/development/contributing_documentation.html#updating-a-pandas-docstring) script for this function?

Done, It's the first error
image

After which it's this
image

Left a line between %(raises_section)s and notes accidentally.
image

**Review by MarcoGorelli — APPROVED** [review]
Nice, thanks!

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @DhruvBShetty

**Pull request merged**